js/interface.js

```diff
@@ -312,8 +318,57 @@ function cacheOriginalEntries(entries, clientIdMap) {
   });
 }
 
+/**
+ * Clear the global timer and hides #alert-live-data by adding a .hidden class
+ * @returns {void}
+ */
+function clearLiveDataTimer() {
+  clearTimeout(globalTimer);
+  $('#alert-live-data').addClass('hidden');
+}
+
+/**
+ * Tracks a global timer and renders the message in State A also shows the warning message
+ * @returns {void}
+ */
+function startLiveDataTimer() {
+  $('#alert-live-data').removeClass('hidden');
+  $('#alert-live-data').html('Modifying data while live users are accessing the app may overwrite data. We recommend using admin screens within the app to modify data safely. \<a target="_blank" href="https://help.fliplet.com">Learn more\</a>');
+
+  globalTimer = setTimeout(function() {
+    $('#alert-live-data').html('Some of the data may have been changed by users of the app or other Studio users. Modifying data while live users are accessing the app may overwrite data. We recommend using admin screens within the app to modify data safely. \<a href="#" data-source-reload>Reload\</a> to see the latest version. \<a target="_blank" href="https://help.fliplet.com">Learn more\</a>');
+
+    Fliplet.Studio.emit('track-event', {
+      category: 'dsm_reload_warning',
+      action: 'show'
+    });
+  }, 300000);
+}
+
+/**
+ * Checks if current app is published or not
+ * @returns {void}
+ */
+function fetchCurrentDataSourcePublishStatus() {
+  getApps.then(function(apps) {
```

Comment:
This isn't how we should be determining whether a data source is connected to a live app. I'll update the tech spec with more information.